src/lib.rs

```diff
@@ -769,6 +773,13 @@ impl Config {
         self.dgram_recv_max_queue_len = recv_queue_len;
         self.dgram_send_max_queue_len = send_queue_len;
     }
+
+    /// Sets the maximum datagram size used by the congestion control.
+    ///
+    /// The default and minimum value is `1200`.
+    pub fn set_max_datagram_size(&mut self, v: usize) {
```

Comment:
Why do we need an additional configuration in addition to `max_udp_payload_size`? The name is also really confusing IMO, considering the name of `max_udp_payload_size` and the fact that we have `DATAGRAM` support now.

Also the documentation seems to be wrong, as the default value is `1452` AFAICT, and we don't enforce any minimum value.

Reply:
My understanding between `max_udp_payload_size` and `max_datagram_size` is two different thing:
- `max_udp_payload_size`: max udp packet size the endpoint can **receive**, telling the peer not to send a packet bigger than this. https://tools.ietf.org/html/draft-ietf-quic-transport-32#section-18.2
- `max_datagram_size`: max packet size considered when the packet is **sent**, once peer's `max_udp_payload_size` is known, it can be additionally limited by that. https://tools.ietf.org/html/draft-ietf-quic-transport-32#section-14
`max_datagram_size` itself comes from the recovery draft: https://tools.ietf.org/html/draft-ietf-quic-recovery-32#section-7.2

Regarding DATAGRAM support, it looks like it's called `max_datagram_frame_size`. Yes those are confusing names so I'm open to the recommendation.

Reply:
But my question still stands, why do we need 2 options? I would expect that in most cases the maximum size we can send and receive is the same, there's not much point in configuring them separately given that in most cases the same value is configured.

Reply:
Path MTU can be asymmetric. Also note that max_udp_payload_size's default is 65527 and max_datagram_size is 1200. For example the sender can send 1200 bytes packets but can receive 1500 bytes packets from the peer. If we combine two api (or meaning of the value) into one, it would result in rejecting incoming 1500 bytes one.

Reply:
Also, I changed the default (MAX_DATAGRAM_SIZE) to 1200 and updated tests.

Reply:
I see what you mean. We could rename the existing `max_udp_payload_size` and call them:
* `max_recv_udp_payload_size` for the existing option (i.e. the transport aprameter)
* `max_send_udp_payload_size` for the new option
?

Reply:
Ok, API renamed as follows. I leave internal variable names unchanged.
- `set_max_udp_payload_size()` -> `set_max_recv_udp_payload_size()`
- `set_max_datagram_size()` -> `set_max_send_udp_payload_size()`